sdk/rust/oak/src/crypto/mod.rs

```diff
@@ -0,0 +1,241 @@
+//
+// Copyright 2019 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Helper library for accessing the Oak cryptography service.
+
+use crate::{grpc, proto::oak::crypto};
+use oak_abi::{
+    label::Label,
+    proto::oak::application::{
+        node_configuration::ConfigType, CryptoConfiguration, NodeConfiguration,
+    },
+};
+use prost::Message;
+use tink_proto::KeyTemplate;
+
+/// Local representation of the connection to an external crypto service.
+pub struct Crypto {
+    client: crypto::OakCryptoClient,
+}
+
+/// Opaque handle for a keyset, which is a collection of key material held
+/// by the Oak crypto pseudo-Node.
+pub type KeysetHandle = u64;
+
+/// Format of an encrypted keyset.
+pub enum KeysetFormat {
+    /// Binary-encoded protobuf.
+    Binary,
+    /// JSON-encoded protobuf.
+    Json,
+}
+
+impl Crypto {
+    /// Creates a [`Crypto`] instance using the given configuration.
+    pub fn new(config: &CryptoConfiguration) -> Option<Crypto> {
+        let config = NodeConfiguration {
+            config_type: Some(ConfigType::CryptoConfig(config.clone())),
+        };
+        crate::io::node_create("crypto", &Label::public_untrusted(), &config)
```

Comment:
This seems limiting if it can only be used to encrypt public data.

Reply:
Bah, used the wrong code as a template &ndash; added a label parameter.  (However, the wider interactions with IFC are still very much [TBD](http://go/oak-crypto#heading=h.txmuvdhg2p5q))